apps/state/lib/state/route_pattern.ex

```diff
@@ -78,4 +88,8 @@ defmodule State.RoutePattern do
     |> Enum.map(& &1.route_pattern_id)
     |> Enum.uniq()
   end
+
+  defp canonical_comparison(_, nil), do: true
```

Comment:
question: is it possible for this value to be `nil`? It doesn't appear to be possible from the types. If it's not possible, then it appears that this function is the same as `==`.

Reply:
From what I can see, there are two existing patterns for empty or otherwise invalid values:
- https://api-v3.mbta.com/routes/?filter[id]= doesn't match anything, and returns an empty list (this is the most common)
- https://api-v3.mbta.com/routes/?page[limit]= ignores the invalid value

There are also cases where we return a bad request error, such as https://api-v3.mbta.com/schedules/?filter[invalid]= but not based on the values that I can see.

Reply:
I think that treating it the same as if nothing was specified is fine. But even with that approach, that's part of the query parameter parsing, not something that should be passed into `State.RoutePattern`: instead, `canonical` should not be a key in the map passed into `filter_by`.